src/utils/cosmosdb_model_versioned.ts

```diff
@@ -149,9 +152,9 @@ export abstract class CosmosdbModelVersioned<
    *  to avoid multi-partition queries.
    */
   public findLastVersionByModelId(
-    modelId: string,
-    partitionKey?: string
+    searchKey: DocumentSearchKey<T, IDENTITY, PK>
```

Comment:
I like the idea but I would like to keep the two params separated, I don't think the array is actually needed:

```ts
public findLastVersionByModelId(
  modelId: DocumentSearchKey<T, IDENTITY, PK>[0],
  partitionKey: DocumentSearchKey<T, IDENTITY, PK>[1]
)
```

Reply:
since partitionKey may be undefined it would be an optional parameter which in part defeat the purpose of this PR which is to use only required parameters once the model is defined

Reply:
I see that this must be a single object

Reply:
If you split parameters you will have a signature like this:
```ts
find(
  modelId: DocumentSearchKey<T, IDENTITY, PK>[0],
  partitionKey: DocumentSearchKey<T, IDENTITY, PK>[1],
  options?: RequestOptions
)
```

Which leads to usages like
```ts
find("abc") // ERROR!
find("abc", undefined) // OK
find("abc", undefined, { /* options */ }) // OK
```

----

> I see that this must be a single object

Do you mean something like this? 
```ts
find({
      modelId,
      partitionKey
  }: DocumentSearchKey<T, IDENTITY, PK>,
  options?: RequestOptions
)
```

I think this leads to a more verbose api
```ts
find({ modelId: "abc", partitionKey: "def" })
```

Reply:
I mean, after I proposed to split the tuple to two params, I checked the rest of the code and realized why you kept it as a single object/tuple

Reply:
What I can do is to consider a scalar instead of a tuple in the case of no partition key
```ts
model.find("abc") // OK
model.find(["abc"]) // ERROR
partitionedModel.find(["abc", "def"]) // OK
partitionedModel.find(["abc"]) // ERROR
partitionedModel.find("abc") // ERROR
```

It's a sugar that isn't worth the effort IMO